src/Jupyter/Extensions.cs

```diff
@@ -174,5 +175,17 @@ public static string Dedent(this string text)
             var leftTrimRegex = new Regex(@$"^[ \t]{{{minWhitespace}}}", RegexOptions.Multiline);
             return leftTrimRegex.Replace(text, "");
         }
+
+        /// <summary>
+        ///      Retrieves and JSON-decodes the value for the given parameter name.
+        /// </summary>
+        public static T DecodeParameter<T>(this Dictionary<string, string> parameters, string parameterName)
+        {
+            if (!parameters.TryGetValue(parameterName, out string parameterValue))
+            {
+                return default(T);
+            }
+            return (T)(JsonConvert.DeserializeObject(parameterValue) ?? default(T));
```

Comment:
I'd suggest adding `#nullable enable` to this file to make sure that you've covered all possible paths for nullable references here.